exasol/error/_parse.py

```diff
@@ -152,134 +155,162 @@ def validate(self, node: ast.Call, file: str) -> Result:
-        if not isinstance(node, ast.List) and not isinstance(node, ast.Constant):
-            self._errors.append(
-                self.Error(
-                    message=self._error_msg.format(
-                        type="mitigations", value=type(node)
-                    ),
-                    file=file,
-                    line_number=node.lineno,
-                )
-            )
+    def _validate_code(self, node: ast.expr, file: str) -> Optional[str]:
+        if code := self._check_node_type(ast.Constant, node, "error-codes", file):
+            return code.value
+        else:
+            return None
+
+    def _validate_message(self, node: ast.expr, file: str) -> Optional[str]:
+        if message := self._check_node_type(ast.Constant, node, "message", file):
+            return message.value
+        else:
             return None
-        if isinstance(node, ast.List):
-            invalid = [e for e in node.elts if not isinstance(e, ast.Constant)]
-            self._errors.extend(
-                [
-                    self.Error(
-                        message=self._error_msg.format(
-                            type="mitigations", value=type(e)
-                        ),
-                        file=file,
-                        line_number=e.lineno,
-                    )
-                    for e in invalid
+
+    def _validate_mitigations(self, node: ast.expr, file: str) -> Optional[List[str]]:
+        if mitigation := self._check_node_types(
+            ast.List, ast.Constant, node, "mitigations", file
+        ):
+            if isinstance(mitigation, ast.List):
+                invalid = [
+                    e for e in mitigation.elts if not isinstance(e, ast.Constant)
                 ]
-            )
-            if invalid:
-                return None
-            else:
-                return [e.value for e in node.elts if isinstance(e, ast.Constant)]
-        elif isinstance(node, ast.Constant):
-            return [node.value]
+                self._errors.extend(
+                    [
+                        Error(code=INVALID_ERROR_CODE_DEFINITION.identifier,
+                              message=INVALID_ERROR_CODE_DEFINITION.message,
+                              mitigations=INVALID_ERROR_CODE_DEFINITION.mitigations,
+                              parameters={"error_element": "mitigations", "file": file, "line": node.lineno, "defined_type": type(e)})
+                        for e in invalid
+                    ]
+                )
+                if invalid:
+                    return None
+                else:
+                    return [
+                        e.value for e in mitigation.elts if isinstance(e, ast.Constant)
+                    ]
+            elif isinstance(mitigation, ast.Constant):
+                return [mitigation.value]
         return None
 
+    def normalize(self, params):
+        for k, v in zip(params.keys, params.keys):
```

Comment:
this still broken, do you plan this in a extra PR

Reply:
Yes, number of LOC changed are already getting out of control